lib/uksignal/signal.c

```diff
@@ -297,14 +286,129 @@ int sigwait(const sigset_t *set, int *sig)
 	ptr->wait.status = UK_SIG_NOT_WAITING;
 
 	/* do not execute handler, set received signal */
-	*sig = ptr->wait.received_signal.si_signo;
+	memcpy(info, &ptr->wait.received_signal,
+	       sizeof(*info));
```

Comment:
```suggestion
	*info = ptr->wait.received_signal;
```

Then you can remove the `include <string.h>`.